File: android/app/src/main/java/com/sqlplay/MeasureSize.java
package com.sqlplay;

import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import java.util.Map;
import java.util.HashMap;

import android.util.Log;
import android.graphics.Typeface;
import android.graphics.Paint;

public class MeasureSize extends ReactContextBaseJavaModule {
   MeasureSize(ReactApplicationContext context) {
       super(context);
   }
    @Override
    public String getName() {
        return "MeasureSizeModule";
    }

    private float getWidthOfString(String string, String fontFamily, float fontSize) {
        Paint paint = new Paint();
        paint.setTypeface(Typeface.create(fontFamily, Typeface.NORMAL));
        paint.setTextSize(fontSize);
        return paint.measureText(string);
    }

    @ReactMethod
    public void measureSize(String text, String font, Integer size, Promise promise) {
        try {
            float width = getWidthOfString(text, font, size);
            promise.resolve(width);
        } catch (Exception e) {
            promise.reject(e);
        }
    }}

File: android/app/src/main/java/com/sqlplay/MeasureSize.kt
package com.sqlplay

import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

import android.graphics.Typeface;
import android.graphics.Paint;

class MeasureSize(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
    override fun getName() = "MeasureSize"

     private fun getWidthOfString(text: String, fontFamily: String, fontSize: Float ): Float {
        val paint = Paint()
        paint.setTypeface(Typeface.create(fontFamily, Typeface.NORMAL))
        paint.textSize = fontSize
        return paint.measureText(text)
    }

    @ReactMethod
    fun measureSize(text: String, font: String, fontSize: Float, promise: Promise) {
        try {
            val width = getWidthOfString(text, font, fontSize);
            promise.resolve(width);
        } catch (e: Exception) {
            promise.reject(e);
        }
    }}
